addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1038 @@
+
+            if fromtype == 'stop' and self.stop_date:
+                self.stop_datetime = self.stop_date
+                self.stop = self.stop_date
+
+    def new_invitation_token(self):
+        return uuid.uuid4().hex
+
+    def create_attendees(self):
+        res = {}
+        CalendarAttendee = new_attendees = self.env['calendar.attendee']
+        ResPartner = self.env['res.partner']
+        for event in self:
+            attendees = event.attendee_ids.mapped("partner_id")
+            for partner in event.partner_ids:
+                if partner in attendees:
+                    continue
+                values = {
+                    'partner_id': partner.id,
+                    'event_id': event.id,
+                    'access_token': self.new_invitation_token(),
+                    'email': partner.email,
+                }
+
+                if partner == self.env.user.partner_id:
+                    values['state'] = 'accepted'
+
+                attendee = CalendarAttendee.create(values)
+                new_attendees += attendee
+                ResPartner += partner
+
+                if not self.env.user.email or self.env.user.email != partner.email:
+                    mail_from = self.env.user.email or tools.config.get('email_from', False)
+                    if not self.env.context.get('no_email'):
+                        attendee._send_mail_to_attendees(email_from=mail_from)
+
+            if new_attendees:
+                event.write({'attendee_ids': [(4, att) for att in new_attendees.ids]})
+            if ResPartner:
+                event.message_subscribe(ResPartner.ids)
+
+            # We remove old attendees who are not in partner_ids now.
+            all_attendee_ids = event.attendee_ids
+            partner_ids_to_remove = (all_attendee_ids.mapped('partner_id') | ResPartner) - event.partner_ids
+            attendee_to_remove = None
+
+            if partner_ids_to_remove:
+                attendee_ids_to_remove = self.env["calendar.attendee"].search([('partner_id.id', 'in', partner_ids_to_remove.ids), ('event_id.id', '=', event.id)])
+                attendee_to_remove = attendee_ids_to_remove
+                if attendee_ids_to_remove:
+                    attendee_ids_to_remove.unlink()
+
+            res[event.id] = {
+                'new_attendee_ids': new_attendees,
+                'old_attendee_ids': all_attendee_ids,
+                'removed_attendee_ids': attendee_to_remove
+            }
+        return res
+
+    def get_search_fields(self, order_fields, r_date=None):
+        sort_fields = {}
+        for ord_field in order_fields:
+            if ord_field == 'id' and r_date:
+                sort_fields[ord_field] = '%s-%s' % (self[ord_field], r_date.strftime("%Y%m%d%H%M%S"))
+            else:
+                sort_fields[ord_field] = self[ord_field]
+                if isinstance(self[ord_field], odoo.models.BaseModel):
+                    name_get = self[ord_field].display_name
```

Comment:
My mistake: display_name will not work if field is o2m or m2m(may have recorset with multiple records), so use name_get and make it as it was before.